### ISSUE TITLE
deep_get cannot index nested arrays

### ISSUE THREAD

**Issue by npow** [dgilland/pydash#61]
Is this supposed to work in pydash?

```
In [4]: o=[{"hits":{"hits":[{"_source":5}]}}]

In [5]: pydash.deep_get(o, '[0].hits.hits[0]._source') # this is supposed to return 5

In [6]: pydash.deep_get(o, '[0].hits.hits')
Out[6]: [{'_source': 5}]

In [7]: pydash.deep_get(o, '[0].hits.hits[0]')

```

It works fine in lodash:
```
> o=[{"hits":{"hits":[{"_source":5}]}}]
[ { hits: { hits: [Object] } } ]
> l.get(o, '[0].hits.hits[0]._source')
5
```

**Comment by dgilland** [comment]
The correct path string that works in pydash is `'[0].hits.hits.[0]._source'`.

```python
>>> pydash.deep_get(o, '[0].hits.hits.[0]._source')
5
```

Pydash requires a `'.'` separator between all keys or indexes.

Supporting `'[0].hits.hits[0]._source'` would require an update to pydash. I'm not opposed to adding this functionality if you want to tackle a PR.

The relevant function to modify is [pydash.objects:path_keys](https://github.com/dgilland/pydash/blob/217c2e6269ac5bfa492adb9a54f39e40643104f3/pydash/objects.py#L1298).

**Comment by dgilland** [comment]
I decided to go ahead and fix this.

Pydash can now support that `key[index]` format:

```
>>> import pydash
>>> o = [{"hits": {"hits": [{"_source": 5}]}}]
>>> pydash.deep_get(o, '[0].hits.hits[0]._source')
5
>>> pydash.deep_get(o, '[0].hits.hits.[0]._source')
5
```

Also as a consequence, things like this work too:

```
>>> o = [{"hits": {"hits": [{"_source": 5}]}}]
>>> pydash.deep_get(o, '[0]hits.hits[0]_source')
5
>>> o = [{'hits': [{'hits': [{'_source': 5}]}]}]
>>> pydash.deep_get(o, '[0]hits[0]hits[0]_source')
5
```

I've released this fix as v3.4.3: https://pypi.python.org/pypi/pydash/3.4.3

Closing the issue for now, but feel free to re-open if you find any other issues related to this.

**Comment by npow** [comment]
Thanks for the quick turnaround!
